openfasoc/generators/gdsfactory-gen/pygen/pdk/sky130_mapped/grules.py

```diff
@@ -0,0 +1,371 @@
+from ..mappedpdk import MappedPDK
+
+grulesobj = dict()
+for glayer in MappedPDK.valid_glayers:
+    grulesobj[glayer] = dict((x, None) for x in MappedPDK.valid_glayers)
+
+grulesobj["dnwell"]["dnwell"] = {"min_width": 3.0, "min_separation": 6.3}
```

Comment:
Are these values manually entered, or are they pulled from some file? If they are pulled from a file from the PDK, can this process also be automated?